### ISSUE TITLE
Create LoadingButton component

### ISSUE THREAD

**Issue by adamstankiewicz** [edx/edx-portal#127]
Now that we have more than one button that includes a loading spinner (e.g., download CSV button, bulk action "Go" button, "Request Codes" button), it makes sense to create a new component `LoadingButton` that can be used across all buttons that have a spinner, e.g.:

```jsx
<LoadingButton
  className="btn-primary"
  loading={isLoading}
  label="My Loading Button"
  onClick={() => { // ... }}
/>
```

This would remove the repeated spinner logic across all buttons that include a spinner.

Once this component is in place, we should also refactor the aforementioned buttons to use `LoadingButton`.

**Comment by wittjeff** [comment]
Accessibility issues to consider:
- Are we assuming that focus stays on the button in loading state (I assume yes)?
- Should we enforce focus trapping while in loading state (I think it depends on what will follow when loading is done, so maybe make it a parameter?)?
- How do we convey state changes (probably add aria-live="polite" to the button).  Investigate whether aria-relevant may be useful (and consider compatibility issues with aria-relevant).
- How do we convey "done" or "ready" state to blind users (cycle to that separate state, use a screen-reader-only span to show it [if no graphic is used], and stay in "done/ready" state until the button takes focus again (user has to leave then come back). At that time, switch to clean/default state.
So there should be three states:  
-- clean / default
-- loading / please wait
-- done / ready
Appropriate terms for these states may also vary with the task.
- Can the user cancel the loading action once it has commenced? If so, enable Esc key to cancel.

**Comment by adamstankiewicz** [comment]
This is no longer relevant as Paragon now has a [`StatefulButton` component](https://edx.github.io/paragon/components/statefulbutton).
